### PR TITLE
{vis}[foss/2022a] matplotlib v3.5.2 w/ Python 3.10.4

### PR DESCRIPTION
**Pull request by sassy-crick**
(created using `eb --new-pr`)
Requires:
- [x] PR #15791 
- [x] PR #15787
- [x] PR #15804
- [x] PR #15789

### CODE REVIEW

**Comment by jfgrimm** [comment]
@sassy-crick can you add the `Pillow` easyconfig to the PR?

**Comment by sassy-crick** [comment]
Apologies for that, this slipped through. I have added all the dependencies now, including the ones which are already merged.

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in total)
himem06.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Platinum 8160 CPU @ 2.10GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/80b6e2bebd528a1c35e0209df8404f5b for a full test report.

**Comment by jfgrimm** [comment]
@boegelbot please test @ generoso

**Review by jfgrimm — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/m/matplotlib/matplotlib-3.5.2-foss-2022a.eb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# avoid that matplotlib downloads and builds its own copies of freetype and qhull
+_fix_setup = "sed -e 's/#system_freetype = False/system_freetype = True/g' "
+_fix_setup += "-e 's/#system_qhull = False/system_qhull = True/g' mplsetup.cfg.template > mplsetup.cfg && "
+
+
+_include_path = "export CPLUS_INCLUDE_PATH=$EBROOTFREETYPE/include/freetype2:${CPLUS_INCLUDE_PATH} && "
+
+exts_list = [
+    ('cycler', '0.11.0', {
+        'sources': '%(name)s-%(version)s.tar.gz',
```
[the remark this answers is not in the document]
Comment (sassy-crick) [reply easybuild/easyconfigs/m/matplotlib/matplotlib-3.5.2-foss-2022a.eb]: Done. Thanks for finding this!

**Comment by bedroge** [comment]
Looks like this already got merged in #15894 in the meantime, so I'm closing this PR.

**Pull request closed without merging**